POSHIT/Example CHT application/malaria-usecase-cht/test/targets/household_members_with_malaria_symptoms.spec.js

```diff
@@ -0,0 +1,104 @@
+const chai = require('chai');
+const expect = chai.expect;
+const TestRunner = require('cht-conf-test-harness');
+const { memberAssessment, memberAssessmentTask, householdMember } = require('../form-inputs');
+const { FORMS, TARGETS, CONTACT_TYPES } = require('../../shared-extras');
+const { DateTime } = require('luxon');
+const harness = new TestRunner();
+
+
+describe('Target: Household Members with Malaria Diagnosis', () => {
+  before(async () => {
+    return await harness.start();
+  });
+  after(async () => {
+    return await harness.stop();
+  });
+  beforeEach(async () => {
+    await harness.clear();
+  });
+  afterEach(() => {
+    expect(harness.consoleErrors).to.be.empty;
+  });
+
+  it('should increment if a malaria diagnosis report is submitted', async () => {
+    //test target loads and is initially zero
+    let householdMembersWithMalaria = await harness.getTargets({ type: TARGETS.MEMBERS_WITH_MALARIA });
+    expect(householdMembersWithMalaria).to.have.property('length', 1);
+    expect(householdMembersWithMalaria[0]).to.nested.include({ 'value.pass': 0, 'value.total': 0, 'value.percent': 0 });
+    //test target increments after form is filled
+    //create a household member
+    let result = await harness.fillContactCreateForm(CONTACT_TYPES.HOUSEHOLD_MEMBER, ...householdMember.ok('male', DateTime.local().minus({ years: 22, months: 2 }).toJSDate()));
+    expect(result.errors).to.be.empty;
+    harness.subject = result.contacts[0];
+    //assess member
+    let householdMemberAssessmentResult = await harness.fillForm(FORMS.MEMBER_ASSESSMENT, ...memberAssessment.showSymptoms);
+    expect(householdMemberAssessmentResult.errors).to.be.empty;
+    //submit malaria diagnosis
+    householdMemberAssessmentResult = await harness.fillForm(FORMS.MEMBER_FOLLOW_UP, ...memberAssessmentTask.followUpMalaria);
+    expect(householdMemberAssessmentResult.errors).to.be.empty;
+    //check that target increments
+    householdMembersWithMalaria = await harness.getTargets({ type: TARGETS.MEMBERS_WITH_MALARIA });
+    expect(householdMembersWithMalaria).to.have.property('length', 1);
+    expect(householdMembersWithMalaria[0]).to.nested.include({ 'value.pass': 1, 'value.total': 1, 'value.percent': 100 });
+    //add a new member and confirm value.percent is 50
+    result = await harness.fillContactCreateForm(CONTACT_TYPES.HOUSEHOLD_MEMBER, ...householdMember.ok('female', DateTime.local().minus({ years: 20, months: 2 }).toJSDate()));
+    householdMembersWithMalaria = await harness.getTargets({ type: TARGETS.MEMBERS_WITH_MALARIA });
+    expect(householdMembersWithMalaria).to.have.property('length', 1);
+    expect(householdMembersWithMalaria[0]).to.nested.include({ 'value.pass': 1, 'value.total': 2, 'value.percent': 50 });
+
+    harness.subject = result.contacts[0];
+    //assess new member
+    householdMemberAssessmentResult = await harness.fillForm(FORMS.MEMBER_ASSESSMENT, ...memberAssessment.showSymptoms);
```

Comment:
This test seems repetitive. We could have used the opportunity to assert that an non-assessment or a an assessmemnt without  malaria does not increment